src/frontend/src/lib/components/tokens/TokensSignedIn.svelte

```diff
@@ -24,6 +24,15 @@
 			($balancesStore?.[tokenId]?.data ?? BigNumber.from(0n)).gt(0n) || displayZeroBalance
 	);
 
+	// We start it as undefined to avoid showing an empty list before the first update.
+	let tokensToDisplay: Token[] | undefined = undefined;
```

Comment:
Is there a particular reason why you are introducing a new state? Why not debouncing the filtering that leads to assing `tokens`?

Reply:
No particular reason: I just wanted to separate what would be the tokens list "raw" and the one that we would display, just for clarity (even if there is not final difference in values in the end). But I can for sure merge, if you want

Reply:
It's maybe cleaner but, not introducing a new state has the advantages to improve the performances as debouncing will reduce the number of filtering.